managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -1720,12 +1726,9 @@ void createNewMetadataLedger() {
             @Override
             public void operationComplete() {
                 // We now have a new ledger where we can write
-                synchronized (pendingMarkDeleteOps) {
-                    flushPendingMarkDeletes();
-
-                    // Resume normal mark-delete operations
-                    STATE_UPDATER.set(ManagedCursorImpl.this, State.Open);
-                }
+                flushPendingMarkDeletes();
```

Comment:
I'm not 100% sure on why the 2 operations where grouped on the same lock, though I'm sure the was some reason :), maybe not a good one. In any case, this is unrelated to the specific change of persisting the individually deleted position and should go into a separate PR.